crates/dock/tests/mutability_check.rs

```diff
@@ -0,0 +1,88 @@
+#[cfg(test)]
+mod tests {
+    use expect_test::expect_file;
+    use regex::Regex;
+
+    fn check(dir_name: &str) {
+        let mut out = Vec::new();
+        let mut err = Vec::new();
+
+        let main_file_contents =
+            std::fs::read_to_string(format!("tests/mutability_check/{dir_name}/main.nail"))
+                .unwrap();
+        // //---stdoutで区切ることで、標準出力と標準エラー出力をテストする
+        // //---stderrは必ず//---stdoutより後に書くことが前提
+        let (main_contents, _) = main_file_contents
+            .split_once("//---stdout")
+            .unwrap_or((&main_file_contents, ""));
+        let mut main_contents = main_contents.trim().to_string();
+
+        match dock::execute(
+            dock::NailExecutablePath::RootFile(
+                std::env::current_dir()
+                    .unwrap()
+                    .join(format!("tests/mutability_check/{dir_name}/main.nail")),
+            ),
+            &mut out,
+            &mut err,
+            false,
+        ) {
+            Ok(_) => (),
+            Err(e) => match e {
+                // 型チェックのエラーはExecutionError::Nailで返るため何もしない
+                // それ以外は型チェック以外のエラーなのでpanicしてテストを失敗させる
+                dock::ExecutionError::Nail => (),
+                dock::ExecutionError::Io(e) => panic!("io error: {e}"),
+                dock::ExecutionError::InvalidRootNailFilePath(e) => {
+                    panic!("invalid root nail file path: {e}")
+                }
+            },
+        }
+
+        fn normalize(stdout_or_stderr: String) -> String {
+            let path_pattern = Regex::new(r"(\[.*?)/crates/").unwrap();
+
+            return stdout_or_stderr
+                .lines()
+                .map(|line| normalize_path(line, &path_pattern))
+                .map(|line| format!("// {}", line))
+                .collect::<Vec<String>>()
+                .join("\n");
+
+            /// 環境が異なってもテストが通るようにパスを正規化します。
+            fn normalize_path(line: &str, path_pattern: &Regex) -> String {
+                path_pattern.replace(line, "[/{nail}/crates/").to_string()
+            }
```

Comment:
**Security Issue**: The regex pattern used in 'normalize_path' function does not escape special characters which might lead to incorrect path normalization and potential security issues if paths are manipulated. <br> **Fix**: Ensure that the regex pattern correctly escapes special characters to prevent misinterpretation. <br> **Code Suggestion**: 
 ```
 -            let path_pattern = Regex::new(r"(\\[.*?)/crates/").unwrap();
 +            let path_pattern = Regex::new(r"(\\\[.*?\\])/crates/").unwrap();
 ```